crates/networking/p2p/types.rs

```diff
@@ -111,11 +213,10 @@ impl RLPDecode for NodeRecord {
 
         // all fields in pairs are optional except for id
         let id_pair = pairs.iter().find(|(k, _v)| k.eq("id".as_bytes()));
-        if let Some((_key, id)) = id_pair {
+        if let Some((_key, _)) = id_pair {
```

Comment:
If neither element of the pair is needed, we can refactor this to use `if id_pair.is_some() { ... }`

Reply:
Even better, instead of finding the pair, given that its elements are not needed we could use `if pairs.iter().any(|(k, _v)| k.eq("id".as_bytes())) { ... } else { ... }`